metabase/metabase
# Obey hidden widget settings in ColumnSettings widgets

Closes https://github.com/metabase/metabase/issues/33377

### Description

Currency settings were showing in the table metadata editor for all numeric fields, not just currency fields. The ColumnSettings component was not obeying the `getHidden` property there. Now they do!

![currencySettings](https://github.com/metabase/metabase/assets/30528226/9a1c0896-851b-40ce-8a06-cc01f1718fa0)


### How to verify

Go to admin > table metadata
- choose any table with numeric columns
- click the cog next to a non-currency numeric field
- go to the formatting tab
- see that currency settings only appear if the display type is set to currency
- see that currency settings always appear for currency fields

### Checklist

- [x] make the date labels consistent (https://metaboat.slack.com/archives/C02H619CJ8K/p1693263985627759?thread_ts=1693248027.703289&cid=C02H619CJ8K)
  - `1/31/2018`
  - `31/1/2018`
  - `2018/1/31`
